installer/common.sh

```diff
@@ -244,3 +246,50 @@ create_webserver_file() {
         -e "s/<BYTEBIN_HOST>/$BYTEBIN_IP:$BYTEBIN_PORT/g" \
         "$@"
 }
+
+################################################################################
+# Self logging
+################################################################################
+
+# Running after variable and function initialization because if any of that
+# stuff fails we are in much deeper troubles...
+
+# Source https://stackoverflow.com/a/62292109/1996022
+if [ "$1" = "--log" ]; then
+    # If the first argument is "--log", shift the arg out and continue
+    shift
+else
+    # Make sure the folder exists
+    mkdir -p "$INSTALLER_LOGS_DIR"
+
+    # Compress the old log
+    if [ -f "$INSTALLER_LOG" ]; then
+        old_log="${INSTALLER_LOG/.log/"_$(date "+%Y-%m-%d_%H-%M-%S").log"}"
+
+        mv "$INSTALLER_LOG" "$old_log"
+        command_exists gzip && gzip -q9 -- "$old_log"
+    fi
+
+    # If run without log, re-run this script within a script command so all
+    # script I/O is logged
+    script -qec "$0 --log ${*@Q}" "$INSTALLER_LOG"
+    exit_code=$?
+
+    # Upload logs on error
+    if [ $exit_code -ne 0 ]; then
```

Comment:
This uploads on error automatically.
Should I maybe ask the user if they want it uploaded?

Reply:
I don't mind, but probably a y/n check will work well